packages/vscode-extension/src/debug/officeTaskHandler.ts

```diff
@@ -0,0 +1,191 @@
+        ;;
+
+        return "";
+    }
+
+    // "source|name|scope"
+    const scope = (task.scope as vscode.WorkspaceFolder)?.uri?.toString() || task.scope?.toString();
+    return `${task.source}|${task.name}|${scope ?? ""}`;
+}
+
+function isOfficeTask(task: vscode.Task): boolean {
+    if (task) {
+        const taskName = task.name;
+        if (Object.values(OfficeTaskName).includes(taskName)) {
+            return true;
+        }
+    }
+    return false;
+}
+
+function isDebugPreLaunchTask(task: vscode.Task): boolean {
+    if (task) {
+        // Debug: PowerPoint Desktop
+        // Debug: Word Desktop
+        // Debug: Excel Desktop
+        if (task.execution && <vscode.ShellExecution>task.execution) {
+            const execution = <vscode.ShellExecution>task.execution;
+            const commandLine =
+                execution.commandLine ||
+                `${typeof execution.command === "string" ? execution.command : execution.command.value} ${(
+                    execution.args || []
+                ).join(" ")}`;
+            if (/npm[\s]+run[\s]+start:desktop -- --app (word|excel|powerpoint)/i.test(commandLine)) {
+                return true;
+            }
+        }
+    }
+
+    return false;
+}
+
+function onDidStartTaskHandler(event: vscode.TaskStartEvent): void {
+    const task = event.execution.task;
+    if (isOfficeTask(task) || isDebugPreLaunchTask(task)) {
+        trackedTasks.add(task.name);
+    }
+}
+
+function onDidEndTaskHandler(event: vscode.TaskEndEvent): void {
+    if (isOfficeTask(event.execution.task) || isDebugPreLaunchTask(event.execution.task)) {
+        trackedTasks.delete(event.execution.task.name);
+    }
+}
+
+function onDidStartTaskProcessHandler(event: vscode.TaskProcessStartEvent): void {
+    if (globalVariables.workspaceUri && isValidOfficeAddInProject(globalVariables.workspaceUri.fsPath)) {
+        const task = event.execution.task;
+        if (task.scope !== undefined && (isOfficeTask(task) || isDebugPreLaunchTask(task))) {
+            allRunningOfficeTasks.set(getTaskKey(task), event.processId);
+        }
+    }
+}
+
+
+function onDidEndTaskProcessHandler(event: vscode.TaskProcessEndEvent): void {
+    const task = event.execution.task;
+    if (task.scope !== undefined && (isOfficeTask(task) || isDebugPreLaunchTask(task))) {
+        allRunningOfficeTasks.delete(getTaskKey(task));
+    }
+}
+
+
+async function onDidStartDebugSessionHandler(event: vscode.DebugSession): Promise<void> {
+    if (globalVariables.workspaceUri && isValidOfficeAddInProject(globalVariables.workspaceUri.fsPath)) {
+        const debugConfig = event.configuration;
+        if (
+            debugConfig &&
+            debugConfig.name &&
+            !debugConfig.postDebugTask
+        ) {
+            if (await commonUtils.checkAndSkipDebugging()) {
+                throw new Error(DebugSessionExists);
+            } else {
+                commonUtils.startLocalDebugSession();
```

Comment:
Will `startLocalDebugSession` be called twice? One from `teamsfxTaskHandler` and another from here.
I would suggest to totally separate the debug session handler from the original teamsfx one, e.g., record office addin's own sessions here.

Reply:
@swatDong  Thanks for your comments. I went through the extension.ts activate function :

export async function activate(context: vscode.ExtensionContext) {
  ........
  if (isTeamsFxProject) {
    activateTeamsFxRegistration(context);
  }

  if (isOfficeAddInProject) {
    activateOfficeDevRegistration(context);
  }
  ......
  }
The teamsfxTaskHandler is registered in activateTeamsFxRegistration, but the officeTaskHandler is registered in activateOfficeDevRegistration. They are mutually exclusive for each other. startLocalDebugSession will not be called twice here.

Reply:
Thanks for the explanation, make sense to me.